axiom/client.py

```diff
@@ -1,21 +1,39 @@
 """Client provides an easy-to use client library to connect to your Axiom
 instance or Axiom Cloud."""
+from logging import getLogger
 from requests_toolbelt.sessions import BaseUrlSession
-
+from requests_toolbelt.utils.dump import dump_response
 from .datasets import DatasetsClient
 
 
+def raise_response_error(r):
+    if r.status_code >= 400:
+        dump = dump_response(r)
+        print(dump.decode("utf-8"))
+    r.raise_for_status()
+
+
 class Client:  # pylint: disable=R0903
     """The client class allows you to connect to your self-hosted Axiom
     instance or Axiom Cloud."""
 
     datasets: DatasetsClient
 
-    def __init__(self, url_base: str, token: str):
+    def __init__(self, url_base: str, token: str, org_id: str = None):
         # Append /api/v1 to the url_base
         url_base = url_base.rstrip("/") + "/api/v1/"
 
+        logger = getLogger()
         session = BaseUrlSession(url_base)
-        session.headers.update({"Authorization": "Bearer " + token})
+        # hook on responses, raise error when response is not successfull
+        session.hooks = {"response": lambda r, *args, **kwargs: raise_response_error(r)}
+        session.headers.update(
+            {"Authorization": "Bearer %s" % token, "Content-Type": "application/json"}
```

Comment:
Can't generally set this to JSON, we also have `application/x-ndjson` and `text/csv`

Reply:
I made a change here, made it set application/json by default, and from ingest method, it could be overriden. would that solution be suitable?

Reply:
Sounds good!